### PR TITLE
Widgets for chat

### PR DESCRIPTION
**Pull request by sieuhuflit**
## Overview

Build UI widgets for chat
- [x] UI widget for direct message
- [x] UI widget for private channel
- [x] UI widget for public channel
- [x] Build drawer for spaces

## Screenshots

| Screenshot |
|--------------|
| Screenshot_20230731-003217|

## Demo 

https://github.com/lemonadesocial/lemonade-flutter/assets/7247063/b2ba07cc-afde-4781-a447-7f0e63f75a15

### CODE REVIEW

File: lib/theme/spacing.dart [hunk 1]
```diff
@@ -1,29 +1,42 @@
 class Spacing {
+  /// Represents spacing of 6.0 pixels.
```
Comment (mthinh) [inline lib/theme/spacing.dart:2]: 👍

**Review by mthinh — CHANGES_REQUESTED** [review]

File: lib/core/presentation/widgets/common/collapsible/collapsible_section_widget.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  _CollapsibleSectionState createState() => _CollapsibleSectionState();
+}
+
+class _CollapsibleSectionState extends State<CollapsibleSection> {
+  bool _isExpanded = true;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    return Theme(
```
Comment (sieuhuflit) [inline lib/core/presentation/widgets/common/collapsible/collapsible_section_widget.dart:21]: This due to `ExpansionTile` pass ThemeData to display label + icon correctly

File: lib/core/presentation/widgets/common/collapsible/collapsible_section_widget.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        data:
+            lemonadeAppDarkThemeData.copyWith(dividerColor: Colors.transparent),
```
Comment (mthinh) [inline lib/core/presentation/widgets/common/collapsible/collapsible_section_widget.dart:23]: I think we should not use fixed dark theme here, let use themeData from context and copywith with a new value

**Review by mthinh — APPROVED** [review]
Nice work 👍

**Pull request merged**